sdk/python/kfp/dsl/_python_component.py

```diff
@@ -47,3 +47,23 @@ def _python_component(func):
     return func
 
   return _python_component
+
+def component():
+  """Decorator for component functions that use ContainerOp.
+  This is useful to enable type checking in the DSL compiler
+
+  Usage:
+  ```python
+  @dsl.component
+  def foobar(model: TFModel(), step: MLStep()):
+    return dsl.ContainerOp()
+  """
+  def _component(func):
+    import inspect
+    fullargspec = inspect.getfullargspec(func)
+    args = fullargspec.args
+    annotations = fullargspec.annotations
+    #TODO: convert and record
```

Comment:
Let's just add the attributes to the function object like I do with other decorators.